### DISCUSSION TITLE
Interface changes too often for no reason, please keep more backwards compatibility.

### DISCUSSION THREAD

**Discussion by nirdeepdub** [PyAV-Org/PyAV#1671] [3. General]
**IMPORTANT:** Be sure to replace all template sections {{ like this }} or your issue may be discarded.

## Desired Behavior

{{ A description of how you think PyAV should behave. }}
I don't see why changing Flags.FAST_SEEK should change to Flags.fast_seek.value what is the advantage here ? Is there a real reason to change this interface!?

## Additional context
Breaking changes.
{{ Add any other context about the problem here. }}

**Comment by WyattBlue** [comment]
The advantage? Well, besides making the code more Pythonic and consistent with PEP 8 (you know, that style guide thing that everyone uses), it provides better type safety and more explicit access to the underlying value. The .value part isn't just there to make your fingers do extra typing - it makes it crystal clear when you're working with the enum member versus its raw value.

Addressing the case, I prefer snake_case over SCREAM_CASE.(mutation is scary, not constants).

If you're worried about breaking changes, ..that's kind of the point of major version updates. They're perfect opportunities to fix historical inconsistencies and modernize interfaces. Would you rather we kept all the legacy quirks forever, just for nostalgia's sake?

**Reply by nirdeepdub** [reply]
Thanks for the detailed reply, IMHO: breaking changes need a real reason, and formatting is not a reason by itself to make a change to an interface. The scream case is much better and is more consistent with the overall interface of using += FLAGS which are additive which is a very C-like interface. The ideal way would have been to create a wrapper that just accepts boolean **kwargs that are in lower case and converts down stream to the write flags.
Since features and bugs are not being fixed into older versions, and had about 3 major version updates in the matter of a week, the numbering plan of the versions is not an excuse. 
 I think you should reconsider that backwards compatibility should be prioritized for existing features.

**Reply by WyattBlue** [reply]
Python and C are different languages, so I think it's okay to have different formatting conventions.

I just don't get the entitlement here. You're not a contributor. No one is funding us. Why are you demanding high quality work from total strangers for nothing in return? 

also:

> 3 major version updates in the matter of a week

This is factually inaccurate. The shortest interval is a month for each major version.

**Reply by nirdeepdub** [reply]
I wasnt demanding, I was discussing obviously because I'm a 'heavy' user, and I wrote a wrapper to PyAV which works amazing :)
https://github.com/deepdub-ai/audiosample

Thanks for the work you put in, I just want to keep up with the latest versions, and it becomes harder as interfaces are broken.

**Reply by WyattBlue** [reply]
> Approximately every 6 months the FFmpeg project makes a new major release [(source)](https://ffmpeg.org/download.html)

PyAV basically has to breakwards compat each time there's a new major version of ffmpeg. Sometimes this requires significant changes to maintain compatibility with FFmpeg's evolving API. The removal of `CodecContext.close()` is an example of this. While the enum name isn't required by this, it is very trivial to work around if you want to multiple PyAV major versions.
